src/lib.rs

```diff
@@ -1137,74 +1140,81 @@ fn write_obus(packet: &mut Write, sequence: &mut Sequence,
             let mut bw1 = BitWriter::<BE>::new(&mut buf1);
             // uleb128()
             let mut coded_payload_length = [0 as u8; 8];
-            let leb_size = aom_uleb_encode(obu_payload_size, &mut coded_payload_length);
+            let leb_size = aom_uleb_encode(obu_payload_size + 1, &mut coded_payload_length);
             for i in 0..leb_size {
                 bw1.write(8, coded_payload_length[i])?;
             }
         }
         packet.write(&buf1).unwrap();
         buf1.clear();
 
+        {
+            let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+            bw1.write_obu_header(OBU_Type::OBU_SEQUENCE_HEADER, obu_extension);
+        }
+        packet.write(&buf1).unwrap();
+        buf1.clear();
+
         packet.write(&buf2).unwrap();
         buf2.clear();
     }
 
     let write_frame_header = fi.num_tg > 1 || fi.show_existing_frame;
+    let mut buf2 = Vec::new();
+    {
+        write_uncompressed_header(&mut buf2, fi)?;
+    }
+    let obu_payload_size = buf2.len() as u64;
+    {
+        let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+        // uleb128()
+        let mut coded_payload_length = [0 as u8; 8];
+        let leb_size = aom_uleb_encode(obu_payload_size + 1, &mut coded_payload_length);
+        for i in 0..leb_size {
+            bw1.write(8, coded_payload_length[i]);
+        }
+    }
+    packet.write(&buf1).unwrap();
+    buf1.clear();
 
-    // TODO: If # of tiles > 1 or show_existing_frame is true,
-    // write Frame Header OBU here.
-    if write_frame_header {
-        write_frame_header_obu_helper(packet, sequence, fi, OBU_Type::OBU_FRAME_HEADER,
-            obu_extension);
+    {
+        let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+        bw1.write_obu_header(OBU_Type::OBU_FRAME_HEADER, obu_extension);
     }
+    packet.write(&buf1).unwrap();
```

Comment:
So, regardless whether it is show_existing_frame, writing OBU_FRAME_HEADER obu and sequence obu here.
In that case, if we enable show_existing_frame, it will write sequence obu, which should not.

Reply:
Yeah I didn't test show existing frame, we don't expose it in any way. I think the sequence header write is valid, just wasteful.